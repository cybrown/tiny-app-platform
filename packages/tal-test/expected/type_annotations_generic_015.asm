main()
  entry:
    FunctionRef       name: map_0
    DeclareLocal      name: map, mutable: false, hasInitialValue: true
    Pop               inBlock: false
    FunctionRef       name: func_1
    Literal           1
    MakeArray
    Literal           0
    MakeRecord
    Local             map
    Call
    Pop               inBlock: false
    FunctionRef       name: map_deep_2
    DeclareLocal      name: map_deep, mutable: false, hasInitialValue: true
    Pop               inBlock: false
    Literal           "a"
    FunctionRef       name: func_3
    Literal           1
    MakeRecord
    Literal           1
    MakeArray
    Literal           0
    MakeRecord
    Local             map_deep
    Call
    Pop               inBlock: false
    FunctionRef       name: map_deep2_4
    DeclareLocal      name: map_deep2, mutable: false, hasInitialValue: true
    Pop               inBlock: false
    Literal           "a"
    FunctionRef       name: func_5
    Literal           "b"
    FunctionRef       name: func_6
    Literal           2
    MakeRecord
    Literal           1
    MakeArray
    Literal           0
    MakeRecord
    Local             map_deep2
    Call
map_0(arg)
  entry:
    Literal           null
    MakeArrayForBlock count: 1
func_1(val)
  entry:
    Literal           null
map_deep_2(arg)
  entry:
    Literal           null
    MakeArrayForBlock count: 1
func_3(val)
  entry:
    Literal           null
map_deep2_4(arg)
  entry:
    Literal           null
    MakeArrayForBlock count: 1
func_5(val)
  entry:
    Literal           null
func_6(val2)
  entry:
    Literal           null
