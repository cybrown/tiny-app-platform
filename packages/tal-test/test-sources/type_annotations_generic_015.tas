fun map(arg: (val: number) => null): null {
    null
}

map((val) => null)

fun map_deep(arg: {a: (val: number) => null}): null {
    null
}

map_deep({a: (val) => null})

fun map_deep2(arg: {a: (val: number) => null, b: (val2: string) => null}): null {
    null
}

map_deep2({a: (val) => null, b: (val2) => null})
