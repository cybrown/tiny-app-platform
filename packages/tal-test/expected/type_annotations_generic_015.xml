<program>
  <declarelocal>
    <function>
      <block>
        <literal />
      </block>
    </function>
  </declarelocal>
  <call>
    <local />
    <function>
      <literal />
    </function>
  </call>
  <declarelocal>
    <function>
      <block>
        <literal />
      </block>
    </function>
  </declarelocal>
  <call>
    <local />
    <record>
      <recordentry>
        <function>
          <literal />
        </function>
      </recordentry>
    </record>
  </call>
  <declarelocal>
    <function>
      <block>
        <literal />
      </block>
    </function>
  </declarelocal>
  <call>
    <local />
    <record>
      <recordentry>
        <function>
          <literal />
        </function>
      </recordentry>
      <recordentry>
        <function>
          <literal />
        </function>
      </recordentry>
    </record>
  </call>
</program>
